packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -512,7 +512,10 @@ class _DropdownRoute<T> extends PopupRoute<_DropdownRouteResult<T>> {
       computedMaxHeight = math.min(computedMaxHeight, menuMaxHeight!);
     }
     final double buttonTop = buttonRect.top;
-    final double buttonBottom = math.min(buttonRect.bottom, availableHeight);
+    double buttonBottom = math.min(buttonRect.bottom, availableHeight);
+    if (buttonRect.height < _kMenuItemHeight) {
+      buttonBottom = buttonRect.bottom - kMaterialListPadding.vertical;
+    }
```

Comment:
why kMaterialListPadding?

Reply:
also should the correction to be applied to bottomLimit instead?
It is confusing that we are adjusting the buttonBottom instead of bottomLimit, and it probably shouldn't be called buttonBottom anymore.

Reply:
` buttonBottom = buttonRect.bottom - kMaterialListPadding.vertical;` 
This is existing code, that adds vertical padding to dropdown menu

Reply:
adjusting `bottomLimit` breaks any other behavior, so I came up with this solution